include/electionguard/collections.h

```diff
@@ -0,0 +1,93 @@
+/// @file collections.h
+#ifndef __ELECTIONGUARD_CPP_COLLECTIONS_H_INCLUDED__
+#define __ELECTIONGUARD_CPP_COLLECTIONS_H_INCLUDED__
+
+#include "export.h"
+#include "status.h"
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief A singly-linked list data structure composed of strings for keys and strings for values.
+ */
+typedef struct eg_linked_list_s eg_linked_list_t;
+
+/**
+* @brief Create a new linked list
+* @param[out] handle A pointer to the `eg_linked_list_t` opaque instance
+                     the caller is responsible for freeing the string.
+* @return eg_electionguard_status_t indicating success or failure
+* @retval ELECTIONGUARD_STATUS_SUCCESS The function was successfully executed
+* @retval ELECTIONGUARD_STATUS_ERROR_BAD_ALLOC The function was unable to allocate memory
+*/
+EG_API eg_electionguard_status_t eg_linked_list_new(eg_linked_list_t **out_handle);
```

Comment:
we always declare a new function, so that we can allocate memory in whatever way we need to.